src/nbl/ext/ImGui/shaders/fragment.hlsl

```diff
@@ -1,9 +1,26 @@
 #include "common.hlsl"
 
-[[vk::combinedImageSampler]][[vk::binding(0, 0)]] Texture2D sampleTexture : register(t0);
-[[vk::combinedImageSampler]][[vk::binding(0, 0)]] SamplerState linearSampler : register(s0);
+[[vk::push_constant]] struct PushConstants pc;
+
+// single separable image sampler to handle all textures we descriptor-index
+[[vk::binding(0,0)]] Texture2D textures[NBL_MAX_IMGUI_TEXTURES];
+[[vk::binding(1,0)]] SamplerState samplerStates[NBL_MAX_IMGUI_TEXTURES];
```

Comment:
not sure if DXC allows spec constants for binding and set, so you'll probably have to codegen the fragment shader

Reply:
it does not indeed (@Przemog1 checked)

Reply:
why are you reviewing very old diff

Reply:
> it does not indeed (@Przemog1 checked)

yes indeed, thats why we codegen all now